server/controllers/appointment.js

```diff
@@ -16,7 +17,32 @@ exports.fetchAppointments = asyncHandler(async (req, res, next) => {
     throw new Error("You do not have any appointments");
   }
 
-  res.status(200).json({ success: { appointments: appointments } });
+  if (type === "all") {
+    res.status(200).json({ success: { appointments: appointments } });
+  } else if (type === "past") {
+    const pastAppointments = [];
+    let currentTime = new Date().getTime();
+
+    for (let key in appointments) {
+      const appointmentTime = new Date(appointments[key].time);
+
+      // Converting our time to the appointment's timezone
+      currentTime = new Date(
+        convertToTimeZone(currentTime, appointments[key].timezone)
+      );
+
+      // If the appointment time is before current time, then push into pastAppointments
+      if (currentTime.getTime() < appointmentTime.getTime())
+        pastAppointments.push(appointments[key]);
+    }
+
+    res.status(200).json({ success: { appointments: appointments } });
```

Comment:
Aren't supposed to return pastAppointments? :-)

Reply:
Ah wow haha, yeah I just changed it! Thank you!